src/DataTable/hooks.jsx

```diff
@@ -27,14 +26,15 @@ export const useSelectionActions = (
 ) => {
   const [{ selectedRows, isEntireTableSelected }, dispatch] = controlledTableSelections;
 
-  const clearSelection = () => {
+  const clearSelection = useCallback(() => {
```

Comment:
This exported function should be memoized so it's not responsible for any erroneous component re-renders in consuming components.